src/Middleware.js

```diff
@@ -6,27 +6,29 @@ import {prefix} from './Prefixer.js'
 
 /**
  * @typedef {{
- * 	children?: Element[],
- * 	length: number,
- * 	props: string[],
- * 	return: string,
- * 	root: Element,
- * 	parent?: Element,
- * 	type: string,
- * 	value: string,
+ * 	 parent?: Element,
+ * 	 children?: Element[] | string,
+ * 	 root: Element,
+ * 	 type: string,
+ * 	 props: string[] | string,
```

Comment:
Similarly, this is apparently a string in some cases. Though all references in the codebase treat it as an array (using methods like `.join()`)

Reply:
Can we put the typedef's(single line) where they are created in the code i.e for the Element typedef we can call it node is it is returned from the `node` function and put it next to the `node` functions return JSDOC. For example in https://github.com/thysultan/stylis.js/blob/eff8dc8a0ca05c3a80054ca57915e8a5dba0ab9e/src/Tokenizer.js#L21:

```
/**
 * @typedef {{parent?: node, children?: node[] | string, root: node, type: string, props: string[] | string, value: string, length: number, return: string, line?: number, column?: number}} node
 * @param {string} value
 * @param {nod} root
 * @param {node?} parent
 * @param {string} type
 * @param {string[]} props
 * @param {node[]} children
 * @param {number} length
 * @return node
 */
```

Same thing for middleware in https://github.com/thysultan/stylis.js/blob/eff8dc8a0ca05c3a80054ca57915e8a5dba0ab9e/src/Middleware.js#L41

```
/**
 * @typedef {(element: node, index?: number, children?: (node | string)[], callback?: middleware) => string | void} middleware
 * @param {middleware[]} collection
 * @return {middleware}
 */
```

Reply:
We want `Node` (or `Element`, whatever we are going to have it called) to be reusable across different declarations, not sure how defining a `@typedef` within the same block as a function signature would work. Even if it would work it kinda blends together for me and I wouldn't call this more readable in the end of a day. Defining different types of nodes (`CommentNode`, `DeclarationNode`, etc) is also worth exploring - so I think keeping them separate (but probably in the same file) would improve readability.

Reply:
>Defining different types of nodes (CommentNode, DeclarationNode, etc)

That might be too much to represent in JSDocs given the constraints of jsdocs and my reluctants to spear head a full-on TypeScript re-write. If it works i think an inline @typedef has the benefit of highlighting where to the type is sourced/created from avoiding indirection; in contrast: in its free-floating form it raises the question of whether it's from global source.